Add shared product cloud docs (#35)
crate/cloud-docs: docs/reference/subscription-plans.rst

Subscription Plans
==================

When signing up for the CrateDB Cloud service, you have a choice of
different subscription plans. Each plan is preconfigured for different use cases, depending on your storage and computation needs. At the same

> time, the plans also offer elasticity, to accompany use case changes.

Reviewer: ```suggestion
time, the plans also offer elasticity to accompany use case changes.
```

Within certain plans, you can horizontally scale the capacity of your
cluster up or down by adding or subtracting nodes. Limited vertical scaling is
also possible with storage scaling (up only). For more information on how to

> do this, please refer to the `scaling guide`_.

Reviewer: ```suggestion
do this, please refer to the `scaling guide`_. You can change plans at any time.
```

.. _crfree:

Free
-----

This plan, also known as CRFREE, is aimed at new users for testing and 
evaluation of CrateDB Cloud. It is perpetually free to use. Every user can 
deploy one free tier cluster in their organization without adding a payment
method. This plan also doesn't consume 
any :ref:`Free Credit <free-trial-budget>` that you may have available.

The cluster resources are limited to one node with up to 2 CPUs, 2 GiB of 
memory, and 8 GiB of storage, which is suitable for basic evaluation purposes.
This plan doesn't offer any kind of scaling.

The CRFREE plan is designed for active testing and evaluation. Clusters under 
this plan are suspended after 7 days of inactivity. After another 7 days, the
cluster is automatically deleted if left suspended.

.. _shared:

Shared
------

> Shared-resource plans are used to deploy clusters that allow for better 
> resource utilization. These clusters share compute and storage resources

Reviewer: ```suggestion
Shared-resource plans are used to deploy clusters that share compute and storage resources
```

I don't think "better resource utilization" is really a thing (except for us!)

Reviewer 2: storage resources are not shared, they are dedicated

Reviewer 2: ```suggestion
Shared-resource plans are used to deploy clusters that share CPU resources with other clusters in this category. Memory and storage are allocated exclusively to each individual cluster.
```

with other clusters in this category. Because of this, they offer a more
cost-effective solution for smaller teams and experimental deployments of
low-traffic applications. They are limited to one node with up to 8 CPUs, 12 
GiB of memory, and 1 TiB of storage. 

The vCPU displayed for each tier is an 'up to' value and is not guaranteed.
Memory and storage are guaranteed. Scaling within the Shared tier is supported.
You can scale from S2 to S12 and vice versa. It is also possible to upgrade

> from Shared plans to Dedicated plans, but not the other way around. Snapshots

Reviewer: Do we use the term "snapshots" or backups in other places? Just for consistency.
